Switch from MSON-AST to Data Struct Refract

danielgtaylor opened the pull request:
This removes all references to the MSON-AST and replaces the example with refracted objects.

smizell commented on README.md:
```diff
@@ -462,17 +450,14 @@ Two supported, feature-equal serialization formats are JSON and YAML:
-                "variable": false
+              "element": "object",
+              "meta": {
+                "name": "<resource-name>"
```
I believe we have removed `name` from `meta` at this point.

danielgtaylor replied:
Hmm, would this use `id` or `title` instead?

pksunkara replied:
`id`.

pksunkara commented on README.md:
```diff
@@ -261,14 +261,11 @@ Two supported, feature-equal serialization formats are JSON and YAML:
-                  "typeSpecification": {
-                    "name": "<sub-type>"
-                  }
+                "element": "object",
```
Related discussion in https://github.com/refractproject/rfcs/pull/10/files#r36736501

pksunkara commented:
This need to wait until the above linked discussion is finalized.

zdne commented:
@danielgtaylor @pksunkara can this be resolved with https://github.com/refractproject/rfcs/pull/10 merged?

pksunkara commented:
Yup. We need to update this PR.

zdne commented:
Ping @danielgtaylor

danielgtaylor commented:
I'll get this updated ASAP and push it up for review. Thanks.

danielgtaylor commented:
I believe I have addressed all the feedback and think that the document is now correct, but my confidence level isn't super high so please double check. I'm starting to think that maybe we want to have a source API Blueprint document / example in this repo and generate the output so we can be sure that:

1. The output is accurate based on the tool people will use.
2. The output matches what we expect.

Maybe it's all moot though with the refract changes. Thoughts?

pksunkara commented on README.md:
```diff
@@ -262,13 +262,13 @@ Two supported, feature-equal serialization formats are JSON and YAML:
+                "content":{
+                  "element": "object",
+                  "attributes": {
+                    "typeAttributes": ["<sub-type>"]
```
Minor but the JSON format is not correct here. Similarly below.

pksunkara commented on README.md:
```diff
@@ -204,8 +204,8 @@ A reference object which is used whenever there is a reference to a [Resource Mo
 
-### Data Structure ([Named Type][])
-Definition of an [MSON][] data structure.
+### Data Structure ([Data Structure Element][])
```
This is wrong.

We should remove this section and just link `Data Structure` in other places directly to [Data Structure](https://github.com/refractproject/refract-spec/blob/master/namespaces/api-description-namespace.md#data-structure-element) in API Description namespace.

pksunkara commented:
Reviewed. Can you also look into the following things?

* [ ] Update version
* [ ] Remove `raw` and `html` from media types

danielgtaylor commented:
@pksunkara I believe I've addressed all feedback. I wrote a quick sample document, ran it through drafter and used similar JSON here. Let me know if there's a better way to write any of it, thanks!

pksunkara commented on README.md:
```diff
@@ -512,13 +502,8 @@ MIT License. See the [LICENSE](LICENSE) file.
-[Type Name]: https://github.com/apiaryio/mson-ast#type-name
-[Type Definition]: https://github.com/apiaryio/mson-ast#type-definition
-[Type Section]: https://github.com/apiaryio/mson-ast#type-section
+[Data Structure Element]: https://github.com/refractproject/refract-spec/blob/master/namespaces/data-structure-namespace.md#data-structure-element-element
```
We don't need this anymore.

pksunkara commented on README.md:
```diff
@@ -204,23 +204,17 @@ A reference object which is used whenever there is a reference to a [Resource Mo
-- `element`: `dataStructure` (fixed, required)
-
 ## Media Types
-The `application/vnd.apiblueprint.ast` is the base media type for API Blueprint AST. An API Blueprint AST with raw Markdown descriptions has the `.raw` suffix whereas version with Markdown descriptions rendered into HTML has the `.html` suffix. The base media type serialization format is specified in the `+<serialization format>` appendix.
```
We still need the last sentence.

pksunkara commented:
This PR is currently waiting on feedback from @zdne regarding the removal of `raw` and `html` from media types. If we do decide to remove them, then they need to removed from `Parse Result.md` too.

pksunkara commented:
Also, parse result spec version should be changed to 2.2 and AST versions in the examples of that file should be updated.

zdne commented:
@pksunkara @danielgtaylor I am up for removing `raw` (et al) as long as the implementation will stay backward compatible (e.g. it will still work with the `Accept: application/vnd.apiblueprint.ast.raw+json` header). Does this clarify?

pksunkara commented on Parse Result.md:
```diff
@@ -86,9 +84,9 @@ Parser Result Media Types inherit from the respective [AST Serialization Type](R
 
```
The line above this line needs to be changed

pksunkara commented on Parse Result.md:
```diff
@@ -62,21 +62,19 @@ Description of a warning from the parser.
-+ [`application/vnd.apiblueprint.ast.*+json`](#json-serialization)
-+ `application/vnd.apiblueprint.ast.*+yaml`
++ [`application/vnd.apiblueprint.ast.json`](#json-serialization)
++ `application/vnd.apiblueprint.ast.yaml`
```
`+yaml`

pksunkara commented on Parse Result.md:
```diff
@@ -62,21 +62,19 @@ Description of a warning from the parser.
 
-+ [`application/vnd.apiblueprint.ast.*+json`](#json-serialization)
-+ `application/vnd.apiblueprint.ast.*+yaml`
++ [`application/vnd.apiblueprint.ast.json`](#json-serialization)
```
`+json`

pksunkara commented on Parse Result.md:
```diff
@@ -62,21 +62,19 @@ Description of a warning from the parser.
 
-+ `application/vnd.apiblueprint.parseresult.raw+json` and `application/vnd.apiblueprint.parseresult.html+json`
-+ `application/vnd.apiblueprint.parseresult.raw+yaml` and `application/vnd.apiblueprint.parseresult.html+yaml`
++ `application/vnd.apiblueprint.parseresult.json`
```
`+json`

pksunkara commented on Parse Result.md:
```diff
@@ -62,21 +62,19 @@ Description of a warning from the parser.
-+ `application/vnd.apiblueprint.parseresult.raw+json` and `application/vnd.apiblueprint.parseresult.html+json`
-+ `application/vnd.apiblueprint.parseresult.raw+yaml` and `application/vnd.apiblueprint.parseresult.html+yaml`
++ `application/vnd.apiblueprint.parseresult.json`
++ `application/vnd.apiblueprint.parseresult.yaml`
```
`+yaml`

pksunkara commented:
A few minor things left.

danielgtaylor commented:
@pksunkara should be ready to go!

pksunkara commented on Parse Result.md:
```diff
@@ -62,33 +62,31 @@ Description of a warning from the parser.
 ### JSON Serialization
 
-`application/vnd.apiblueprint.parseresult.*+json; version=2.1`
+`application/vnd.apiblueprint.parseresult+json; version=2.1`
```
You forgot to change the version here.

pksunkara commented:
Just a small minor thing still left.

pksunkara commented:
Review passed. Waiting for drafter release to merge this PR.

pksunkara commented:
Please tag and release
